### PR TITLE
#36884: Signals passed up for note arrival and entity loading.

### PR DESCRIPTION
**Pull request by thebeeland**
Plus access to note thread data as a property of activity stream and version details widgets.

### CODE REVIEW

File: python/activity_stream/data_manager.py [hunk 1]
```diff
@@ -86,6 +86,7 @@ class ActivityStreamDataHandler(QtCore.QObject):
     
     update_arrived = QtCore.Signal(list)
     note_arrived = QtCore.Signal(int, int)
+    note_thread_arrived = QtCore.Signal(int, object)
     thumbnail_arrived = QtCore.Signal(dict)
```
Comment (robblau) [inline python/activity_stream/data_manager.py]: Signals should be documented... when they are emitted and what the arguments are.

File: python/version_details/version_details.py [hunk 2]
```diff
@@ -60,9 +60,13 @@ class VersionDetailsWidget(QtGui.QWidget):
     # along.
     entity_created = QtCore.Signal(object)
 
+    # Emitted when an entity is loaded in the panel.
```
Comment (robblau) [inline python/version_details/version_details.py]: Argument should be doc'ed.

File: python/activity_stream/activity_stream.py [hunk 3]
```diff
@@ -145,6 +146,14 @@ def destroy(self):
     # properties
 
     @property
+    def note_threads(self):
+        """
+        The currently loaded note threads, keyed by Note entity id and
+        containing a list of Shotgun entity dictionaries.
```
Comment (manneohrstrom) [inline python/activity_stream/activity_stream.py]: What are these note threads? Are they all the note threads that are expanded within the activity stream? Please extend the comment to give more context. Maybe also update the  class level doc string.

File: python/activity_stream/activity_stream.py [hunk 4]
```diff
@@ -165,8 +174,7 @@ def _set_allow_screenshots(self, state):
         self._allow_screenshots = bool(state)
         self.ui.note_widget.allow_screenshots(self._allow_screenshots)
 
-    allow_screenshots = QtCore.Property(
-        bool,
+    allow_screenshots = property(
```
Comment (manneohrstrom) [inline python/activity_stream/activity_stream.py]: please add docstring

File: python/activity_stream/activity_stream.py [hunk 5]
```diff
@@ -185,8 +193,7 @@ def _set_show_sg_stream_button(self, state):
         """
         self._show_sg_stream_button = bool(state)
 
-    show_sg_stream_button = QtCore.Property(
-        bool,
+    show_sg_stream_button = property(
```
Comment (manneohrstrom) [inline python/activity_stream/activity_stream.py]: please add docstring

**Pull request merged**